app/forms/fields/date_field.py

```diff
@@ -33,8 +34,8 @@ def __init__(self, validators, **kwargs):
         form_class = get_form_class(validators)
         super().__init__(form_class, **kwargs)
 
-    def process(self, formdata, data=None):
-        if data is not None:
+    def process(self, formdata, data=None, extra_filters=None):
+        if data is not None and not isinstance(data, UnsetValue):
```

Comment:
`data` is now an `UnsetValue`, not None, but unsure if it's necessary to remove the `not None` check